custom-metrics-stackdriver-adapter/pkg/adapter/translator/utils/filter_builder.go

```diff
@@ -231,27 +230,20 @@ func (fb FilterBuilder) WithPods(pods []string) FilterBuilder {
 
 // WithNodes adds a filter for nodes.
 //
-// Note:
-//
-//	for prometheus metrics, their instance information consist of a node name and a port number
-//	(such as "gke-test--default-pool-cee13989-qsky:8080"), but when you use this method, you only
-//	need to provide target node names (such as "gke-test--default-pool-cee13989-qsky"), because we
-//	have "(:\\d+)*" at the end of the regex to ignore the port number part.
-//
 // Example:
 //
-//	// To filter for instances gke-test--default-pool-cee13989-0i75 and
-//	// gke-test--default-pool-cee13989-qsky:8080 with the query
-//	// resource.labels.instance = monitoring.regex.full_match("^" +
-//	// 	"(gke-test--default-pool-cee13989-0i75|gke-test--default-pool-cee13989-qsky)" +
-//	// 	"(:\\d+)*"
```

Comment:
Can you explain this change in query syntax?

Reply:
originally I was trying to filter something in format `node-name:5000` for resource.labels.instance,
1. "^", first part marks start of node name
2. "(node1|node2)", second part is saying it should be either `node1` or `node2`
3. "(:\\d+)*)", third part is saying it's possible to followed by a port number `:5000` (0 or mulultiple), but it cannot be only `:`
since we changed to use metric.labels.node, it will not contain `:5000`, we don't need the third part any more, and in later commits it was changed to `$` which marks the end of name.